packages/server/src/lib/access-helpers.js

```diff
@@ -128,6 +129,7 @@ async function isMicrosoftSafeLinksRequest(req, res, next) {
     const userAgent = req.headers['user-agent'] || '';
     const nativeHost = req.headers['x-native-host'] || '';
     if (userAgent.toLowerCase().includes('oneoutlook') || nativeHost.toLowerCase().includes('oneoutlook')) {
+        log.info({ headers: req.headers }, 'Microsoft Safe Links request');
```

Comment:
Note: if we release this change to production and not see any evidence of this log then it is clear that we have not truly identified what a request from safe-links looks like. If that is the case I'll go ahead and add additional logging around the whole request to get better data.